request-handler/src/main/java/au/org/aodn/aws/wps/lambda/WpsLambdaRequestHandler.java

```diff
@@ -0,0 +1,146 @@
+
+        AmazonS3Client s3Client = (credentials == null) ? new AmazonS3Client() : new AmazonS3Client(credentials);
+        Region region = Region.getRegion(Regions.fromName(s3RegionName));
+        s3Client.setRegion(region);
+
+        String envName = getEnvironmentName(context);
+
+        //  Append an environment name to the front of the config filename
+        if(envName != null)
+        {
+            configFilename  = envName + "/" + configFilename;
+        }
+
+        LOGGER.log("S3 Config file location: " + configFilename);
+
+        S3Object configFile = s3Client.getObject(bucketName, configFilename);
+        S3ObjectInputStream contentStream = configFile.getObjectContent();
+        Properties config = new Properties();
+        try
+        {
+            //  Load configuration from S3 file
+            config.load(contentStream);
+
+            //  Execute the request
+            response = handler.handleRequest(request, config);
+
+        }
+        catch(IOException ioex)
+        {
+            //  Bad stuff happened
+            LOGGER.log("Exception running WPS Lambda function: " + ioex.getMessage());
+        }
+
+        return response;
+    }
+
+    private String getEnvironmentName(Context context)
+    {
+        String envName = null;
+
+        //  If an alias is set for the function version - then it will appear as the last part of the
+        //  InvokedFunctionArn property of the context object.  If no alias is set, then the last
+        //  part of the property will be the functionName.
+        //
+        //  Example -
+        //    With an alias of 'DEV' for the function version invoked:
+        //          arn:aws:lambda:us-east-1:123456789012:function:helloStagedWorld:DEV
+        //
+        //    With no alias set:
+        //          arn:aws:lambda:us-east-1:123456789012:function:helloStagedWorld
+        //
+        //  We will use the alias name (if it exists) to indicate the environment the code is running in
+        //  and hence which configuration file is loaded.
+        int lastColonPosition = context.getInvokedFunctionArn().lastIndexOf(":");
+        String lastStringSegment = context.getInvokedFunctionArn().substring(lastColonPosition + 1);
+
+        LOGGER.log("Function name        : " + context.getFunctionName());
+        LOGGER.log("Invoked function ARN : " + context.getInvokedFunctionArn() );
+        if(!context.getFunctionName().equalsIgnoreCase(lastStringSegment))
+        {
+            //  There must be an alias
+            envName = lastStringSegment;
+        }
+        else
+        {
+            //  No alias passed - use a default
+            envName = DEFAULT_ENV_NAME;
+        }
+
+        return envName;
+    }
+
+    private String getEnvironmentVariable(String keyName)
+    {
+        return System.getenv(keyName);
+    }
+
+
+    private String getEncryptedEnvironmentVariable(String keyName)
```

Comment:
I was looking into encrypting the access key and secret key environment variables - so they wouldn't be in clear text in the management console.  If there are encrypted environment variables - this is how you read them at runtime.